src/UnitSystem.cc

```diff
@@ -29,57 +29,49 @@
 #include "lsst/afw/geom/transformFactory.h"
 #include "lsst/meas/modelfit/UnitSystem.h"
 
-namespace lsst { namespace meas { namespace modelfit {
+namespace lsst {
+namespace meas {
+namespace modelfit {
 
-UnitSystem::UnitSystem(afw::geom::SpherePoint const & position,
-                       std::shared_ptr<const lsst::afw::image::Calib> calibIn,
-                       double flux) {
-    auto scale = 1.0*lsst::afw::geom::arcseconds;
+UnitSystem::UnitSystem(afw::geom::SpherePoint const& position,
+                       std::shared_ptr<const lsst::afw::image::PhotoCalib> photoCalib_, double flux) {
+    auto scale = 1.0 * lsst::afw::geom::arcseconds;
     auto cdMatrix = afw::geom::makeCdMatrix(scale);
     wcs = afw::geom::makeSkyWcs(afw::geom::Point2D(0.0, 0.0), position, cdMatrix);
-    calibIn = validateCalib(calibIn);
-    Scalar mag = calibIn->getMagnitude(flux);
-    PTR(afw::image::Calib) calib_ = std::make_shared<afw::image::Calib>();
-    calib_->setFluxMag0(std::pow(10.0, mag/2.5));
-    calib = calib_;
+    photoCalib_ = validatePhotoCalib(photoCalib_);
+    Scalar mag = photoCalib_->instFluxToMagnitude(flux);
+    photoCalib = afw::image::makePhotoCalibFromCalibZeroPoint(std::pow(10.0, mag / 2.5), 0.0);
 }
 
-UnitSystem::UnitSystem(afw::geom::SpherePoint const & position, Scalar mag) {
-    auto scale = 1.0*lsst::afw::geom::arcseconds;
+UnitSystem::UnitSystem(afw::geom::SpherePoint const& position, Scalar mag) {
+    auto scale = 1.0 * lsst::afw::geom::arcseconds;
     auto cdMatrix = afw::geom::makeCdMatrix(scale);
     wcs = afw::geom::makeSkyWcs(afw::geom::Point2D(0.0, 0.0), position, cdMatrix);
-    PTR(afw::image::Calib) calib_ = std::make_shared<afw::image::Calib>();
-    calib_->setFluxMag0(std::pow(10.0, mag/2.5));
-    calib = calib_;
+    photoCalib = afw::image::makePhotoCalibFromCalibZeroPoint(std::pow(10.0, mag / 2.5), 0.0);
 }
 
-std::shared_ptr<const lsst::afw::image::Calib> UnitSystem::validateCalib(
-    std::shared_ptr<const lsst::afw::image::Calib> calib_) {
-    if (calib_->getFluxMag0().first == 0.0){
-        return getDefaultCalib();
-    } else{
-        return calib_;
+std::shared_ptr<const lsst::afw::image::PhotoCalib> UnitSystem::validatePhotoCalib(
+        std::shared_ptr<const lsst::afw::image::PhotoCalib> photoCalib_) {
```

Comment:
Another underscore in the API.

Reply:
Consider a `using lsst::afw::image::PhotoCalib;` declaration at the head of the file to make the code a bit less verbose.

Reply:
I removed the `lsst::` from it, which helps.